### PR TITLE
Retrieve road before creating from duplicate

### LINKED ISSUE #331: duplicating a road which hasn't been loaded yet
If you create a new road and copy one that hasn't been loaded yet (because you haven't clicked that tab), the new road is blank. There are no errors and it still works fine after you've loaded the appropriate road, so I don't think this is a big deal. The workaround for users is easy and natural--click the original tab, confirm nonempty, try again.

### PR DESCRIPTION
**Pull request by georgiashay**
Fixes #331

### CODE REVIEW

File: src/components/Auth.vue [hunk 1]
```diff
@@ -252,6 +252,9 @@ export default {
           road: roadData.data.file,
           ignoreSet: true
         });
+
+        _this.$store.commit('setRetrieved', roadID);
```
Comment (npfoss) [inline src/components/Auth.vue]: this line missing seems like it should have caused more problems...

**Comment by georgiashay** [comment]
sorry, am dumb

You will probably need to clear your cookies for courseroad to test this again, the buggy road won't delete itself.

**Comment by georgiashay** [comment]
Do you mean road retrieval should be in the store?  I might agree but right now all of our auth/saving/retrieving related stuff is in Auth.vue, so I'm not sure if we'd want to do some sort of refactor here?

File: src/components/RoadTabs.vue [hunk 2]
```diff
@@ -149,16 +161,22 @@ export default {
...
+            this.roads[this.duplicateRoadSource].contents.coursesOfStudy.slice(0),
+            this.roads[this.duplicateRoadSource].contents.selectedSubjects.map((semester) => semester.slice(0)),
+            Object.assign({}, this.roads[this.duplicateRoadSource].contents.progressOverrides)
+          );
+          this.addDialog = false;
+          this.newRoadName = '';
+        }
       }
-      this.addDialog = false;
-      this.newRoadName = '';
```
Comment (npfoss) [inline src/components/RoadTabs.vue:161]: why duplicate this at the end of both if/else branches?

File: src/components/RoadTabs.vue [hunk 3]
```diff
@@ -149,16 +161,22 @@ export default {
...
-          Object.assign({}, this.roads[this.duplicateRoadSource].contents.progressOverrides)
-        );
+        if (this.$store.state.unretrieved.indexOf(this.duplicateRoadSource) >= 0) {
+          this.$emit('retrieve', this.duplicateRoadSource);
+        } else {
+          this.$emit('add-road',
+            this.newRoadName,
+            this.roads[this.duplicateRoadSource].contents.coursesOfStudy.slice(0),
+            this.roads[this.duplicateRoadSource].contents.selectedSubjects.map((semester) => semester.slice(0)),
+            Object.assign({}, this.roads[this.duplicateRoadSource].contents.progressOverrides)
```
Comment (npfoss) [inline src/components/RoadTabs.vue:174]: seems like an exact copy of the `$store.state.unretrieved` watcher. is there a nice way to combine them?

File: src/components/RoadTabs.vue (hunk 2 above)
Comment (georgiashay) [reply src/components/RoadTabs.vue:161]: it's only in 2/3 branches; it doesn't happen if you need to retrieve the road

Comment (npfoss) [reply src/components/RoadTabs.vue:161]: oh oops, you're totally right

**Pull request merged**